backend/ethereum/channel/test/simulated.go

```diff
@@ -122,17 +122,22 @@ func (s *SimulatedBackend) FundAddress(ctx context.Context, addr common.Address)
 	if err != nil {
 		panic(err)
 	}
-	tx := types.NewTransaction(nonce, addr, test.MaxBalance, GasLimit, big.NewInt(GasPrice), nil)
-	signer := types.NewEIP155Signer(params.AllEthashProtocolChanges.ChainID)
-	signedTX, err := types.SignTx(tx, signer, s.faucetKey)
+	txdata := &types.DynamicFeeTx{
```

Comment:
Why use the concrete type `DynamicFeeTx`? Wouldn't `types.NewTransaction` automatically give the correct type?

Reply:
`NewTransaction` has the comment
```
// NewTransaction creates an unsigned legacy transaction.
// Deprecated: use NewTx instead.
```
And `NewTx` takes any of `LegacyTx`, `DynamicFeeTx` or `AccessListTx` (no idea what the 3. is).

Note that I use `types.SignNewTx` below, which is just a shortcut for first calling `NewTx` followed by `SignTx`.

Reply:
It made me realize that we still use `types.NewTransaction` in `test/transactor.go`. Will also change them to the new tx types.